### ISSUE TITLE
test_hist_impl fails in Release mode

### ISSUE THREAD

**Issue by psteinb** [sqeazy/sqeazy#30]
on a 2x12c _E5-2670 v3 @ 2.30GHz_
```
$ ./tests/test_hist_impl 
Running 19 test cases...
/home/steinbac/development/sqeazy/src/cpp/tests/test_hist_impl.cpp(195): error: in "parallel_hist_impl_unsigned/mode_and_mean": check of_variable.mode() == 170 has failed [0xa1 != 0xaa]

*** 1 failure is detected in the test module "TEST_HIST_IMPL"
```

**Comment by psteinb** [comment]
bug reproduced with debug symbols as well

**Comment by psteinb** [comment]
fixed with b83874d
